src/main/java/teammates/ui/webapi/CalculateUsageStatisticsAction.java

```diff
@@ -0,0 +1,88 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+import teammates.common.datatransfer.QueryLogsResults;
+import teammates.common.datatransfer.attributes.UsageStatisticsAttributes;
+import teammates.common.datatransfer.logs.GeneralLogEntry;
+import teammates.common.datatransfer.logs.LogEvent;
+import teammates.common.datatransfer.logs.LogSeverity;
+import teammates.common.datatransfer.logs.QueryLogsParams;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Logger;
+import teammates.common.util.TimeHelper;
+
+/**
+ * Gathers usage-related statistics (e.g. new created entities) in the past hour and store in the database.'
+ */
+public class CalculateUsageStatisticsAction extends AdminOnlyAction {
```

Comment:
Would `CalculateHourlyUsageStatisticsAction` be better?

Reply:
As I mentioned, I want to keep the option open for when we want to gather statistics for different time period. With that said, I shouldn't write "past hour" in the class docs, and that is what I'm going to change.